pkg/frontend/admin_openshiftcluster_vmresize.go

```diff
@@ -45,44 +43,37 @@ func (f *frontend) _postAdminOpenShiftClusterVMResize(log *logrus.Entry, ctx con
 		return err
 	}
 
-	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	// fetches all the Virtual machines of the Cluster RG
+	vmList, err := action.ListVMsOfResourceGroup(ctx)
 	if err != nil {
 		return err
 	}
+	filteredVmList := f.filterVMList(vmList)
 
-	nodeList, err := k.KubeList(ctx, "node", "")
-	if err != nil {
-		return err
-	}
-
-	var u unstructured.Unstructured
-	var nodes corev1.NodeList
-	if err = json.Unmarshal(nodeList, &u); err != nil {
-		return err
+	// checks if the provided VMname is under the RG
+	if !stringutils.Contains(filteredVmList, vmName) {
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeNotFound, "",
+			`"The VirtualMachine '%s' under resource group '%s' was not found."`,
+			vmName, resourceGroupName)
 	}
 
-	err = kruntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &nodes)
-	if err != nil {
-		return err
+	// bland check, to validate if the node is master by checking the vmName has substring 'master'
+	if !strings.Contains(strings.ToLower(vmName), "master") {
```

Comment:
nit: would it be a little better to use regex matching like .*-master-[0-9]{1,}, so for corner cases that the cluster name contains "master", the worker nodes wont be allowed.

Reply:
Done.